tracing-serde/src/lib.rs

```diff
@@ -156,20 +186,26 @@ impl<'a> Serialize for SerializeRecord<'a> {
         S: Serializer,
     {
         let serializer = serializer.serialize_map(None)?;
-        let mut visitor = SerdeMapVisitor {
-            serializer,
-            state: Ok(()),
-        };
+        let mut visitor = SerdeMapVisitor::new(serializer, Ok(()));
         self.0.record(&mut visitor);
         visitor.finish()
     }
 }
 
-struct SerdeMapVisitor<S: SerializeMap> {
+pub struct SerdeMapVisitor<S: SerializeMap> {
```

Comment:
Nit: if this is `pub` now, can it have a doc comment + derive `Debug`?

Reply:
What kind of documentation you're thinking of? Didn't really write it myself, so the best I can think of is

> A visitor to go through a map of values.

Reply:
Also I'd make the more of the visitors public in here and document them. Although it's not really in the scope of this PR anymore!

Reply:
Sure, I agree that API changes in `tracing-serde` are out of scope for this branch. I'd like to make sure that any public APIs are documented before publishing that crate, though, which will be a blocker for getting these `tracing-subscriber` changes published to crates.io (a crate with unpublished dependencies cannot be published).

Reply:
As for a doc comment, what about:
```rust
/// A [field visitor] that serializes each visited field to a `serde` [`SerializeMap`].
///
/// This type is used when the fields on a `tracing` [span] or [event] should 
/// be serialized as a map of key-value pairs.
///
/// [field visitor]: https://docs.rs/tracing/0.1.9/tracing/field/trait.Visit.html
/// [`SerializeMap`]: https://docs.rs/serde/1.0.101/serde/ser/trait.SerializeMap.html
/// [span]: https://docs.rs/tracing/0.1.9/tracing/span/index.html
/// [event]: https://docs.rs/tracing/0.1.9/tracing/struct.Event.html
```

Reply:
Seems like I could make this private again.